File: stacktrace/net9/Services/StacktraceService.cs
using System.Diagnostics;

namespace net9.Services;

public class StacktraceService: IStacktraceService
{
    public string EnvirontmentStacktrace()
    {
        string stacktrace = Environment.StackTrace;
        Console.WriteLine(stacktrace);
        return "environtment stacktrace";
    }

    public string ExceptionStacktrace()
    {
        try
        {
            throw new Exception("error happen");
        }
        catch(Exception e)
        {
            Console.WriteLine(e.StackTrace);
            return "error";
        }
    }

    public string PrintStacktrace()
    {
        StackTrace stacktrace = new StackTrace(true);
        Console.WriteLine(stacktrace.ToString());
        return "ok";
    }
}

File: stacktrace/rust/src/services/stacktrace_service.rs
use std::backtrace::Backtrace;

pub trait StacktraceService {
    async fn printStacktrace(&self) -> String;
}

pub struct StacktraceServiceImpl {

}

impl StacktraceServiceImpl {
    pub fn new() -> StacktraceServiceImpl {
        StacktraceServiceImpl{

        }
    }
}

impl StacktraceService for StacktraceServiceImpl {
    async fn printStacktrace(&self) -> String {
        let bt = Backtrace::capture();
        println!("{}", bt);
        "ok".to_string()
    }
}